TSOClient/tso.client/UI/Panels/UILotPurchaseDialog.cs

```diff
@@ -15,8 +15,11 @@ namespace FSO.Client.UI.Panels
 {
     public class UILotPurchaseDialog : UIDialog
     {
-        private Regex VALIDATE_NUMERIC = new Regex(".*[0-9]+.*");
-        private Regex VALIDATE_SPECIAL_CHARS = new Regex("[a-z|A-Z|-| |']*");
+        private Regex VALIDATE_NUMERIC = new Regex("[0-9]");
+        private Regex VALIDATE_SPECIAL_CHARS = new Regex(@"[^\p{L} '-]");
+        private Regex VALIDATE_APOSTROPHES = new Regex("^[^']*'?[^']*$");
+        private Regex VALIDATE_DASHES = new Regex("^[^-]*-?[^-]*$");
+        private Regex VALIDATE_SPACES = new Regex("^[^ ]+( [^ ]+)?( [^ ]+)?$");
```

Comment:
Same as above.

Reply:
I don't know why have I done it that way… But it's fixed now. I've also prevented the regex from creating unnecessary groups.